### ISSUE TITLE
Enhancement: Improve the accessibility of the site header

### ISSUE THREAD

**Issue by awtj8o81ryywg793** [matrix-org/matrix.org#3115]
Currently, there are a few issues with the site's header:

- `<label>` elements are used as buttons but do not contain the required attributes (e.g., `role`, `aria-pressed`) to mark them as such.
  - A minimal amount of JavaScript will be required to emulate button functionality.
- `<label>` elements cannot be navigated using the keyboard.
  - Adding the attribute `tabindex=0` would fix this and allow all users to navigate to and from it with ease.
- `<a>` elements of first-level navigation items are child elements of toggles (i.e., they are contained within a `<label>` element), rather than sibling elements.
  - Such a structure seems a little misguided. Even if it doesn't cause issues, at least for mouse users.
- `<label>` elements are missing interactive visuals (e.g., `cursor: pointer`) that demonstrate their button-like functionality.

**Comment by awtj8o81ryywg793** [comment]
~Since starting work on this enhancement (see #3130), I've come to discover further issues beyond what was originally mentioned in the first comment of this issue. To make it easier to see what I suggest adding or refactoring, I'll update the original comment instead of creating an endless list of comments.~

I've revised this decision; see this [comment](https://github.com/matrix-org/matrix.org/pull/3130#issuecomment-3770939706) for more information.
